docs/RFCS/20200811_non_blocking_txns.md

```diff
@@ -0,0 +1,1027 @@
+talk about replication factors and fault tolerance. Probably for the better,
+because these issues are hard to reason about if you don't spend your days
+counting quorums. This point shows the problem with doing such replication
+schemes at the SQL level: we hide from the system the redundancy between the
+indexes (and between the replicas of different indexes), and so then we need to
+pay a lot to duplicate storage and replication work just to maintain the
+independent availability of each index.
+
+# Working Assumptions
+
+The bulk of this proposal is founded on the idea that at some time in the
+future, we will be able to dramatically reduce the clock uncertainty that we
+feel comfortable running with, either across the board or at least within
+CockroachCloud.
+
+Wide area network (WAN) round-trip time (RTT) latencies can be as large as 200ms
+in the global deployment scenarios that we are interested in (ref:
+[AWS](https://docs.aviatrix.com/_images/inter_region_latency.png),
+[Azure](https://docs.aviatrix.com/_images/arm_inter_region_latency.png),
+[GCP](https://docs.aviatrix.com/_images/gcp_inter_region_latency.png)). The
+current default clock uncertainty offset is 500ms. This proposal explores the
+half of the design space in which clock uncertainty bounds drop below WAN RTT
+time. Within this regime, it becomes cheaper to establish causality between
+events using clocks and waiting than by communicating with distant nodes.
+
+As such, the uncertainty bounds necessary for this proposal to make sense are
+much lower than what we currently use, but are also well within the realm of
+possibility for software-only solutions. Clouds are continuing to invest into
+the software-level time synchronization services that they provide. For
+instance, Amazon released a [Time Sync
+Service](https://aws.amazon.com/about-aws/whats-new/2017/11/introducing-the-amazon-time-sync-service/)
+in 2017 that strives to provide reliable time references for EC2 instances.
+Similarly, software-level time synchronization has seen interest from the
+academic community in recent years [4], with research even leading to
+[productionized service offerings](https://www.ticktocknetworks.com/) in some
+cases. Lastly, users are already running CockroachDB with battle-tested time
+synchronization services like [Kronos](https://github.com/rubrikinc/kronos)
+which we may eventually explore using or extending.
+
+All of this means that dramatically reducing the clock uncertainty bounds that
+we run with seems realistic over the next few years. Combined with that fact
+that this proposal makes sense even for fairly large clock uncertainty values
+(up to 100ms), it is safe to say that atomic clocks and TrueTime are not
+prerequisites for this proposal!
+
+For the rest of this proposal, we will assume that the clock uncertainty offset
+is configured to 30ms. We will also assume that the global cluster topologies
```

Comment:
Will this proposal still work if the offset is configured to the current offset of 500ms? Will it just force the writers to wait longer?